src/components/MapController/MapController.js

```diff
@@ -150,9 +150,23 @@ var MapController = React.createClass({
           onSetArea={this.handleSetArea}
           area={this.state.area}
         />
+      );    
+      var dataAttributes = this.state.dataset.split('_');
```

Comment:
We *really* don't want to use the dataset ID to extract metadata. This builds weird assumptions into our code that shouldn't be there (and means that the rules around dataset ID are too rigid, and *have* to be followed, else break the application). Dataset ID should be assumed to contain nothing more than a unique sequence of characters... it could just as well be a monotonically increasing number or a hexadecimal hash.

With that said, could you refactor this to use the information from the `medatadata` and `multimeta` API calls instead? The information should have already been retrieved by the front-end by this point in the code.